### ISSUE TITLE
[BUG] Upload of images fails on Oneplus 6

### ISSUE THREAD

**Issue by Kraksna83** [owncloud/android#2919]
### Actual behaviour
-Upload from android app fails. 

### Expected behaviour
-The upload should succeed or produce error message. 
 
### Steps to reproduce
1. Install latest server on fully updated stable debian 
2. Install android app from google play. 
3. Upload a random file ( image, in this case )*

*it appears that uploading the first file worked well.  Its the images which fail 100% of the time.  

Can this problem be reproduced with the official owncloud server?
(url: https://demo.owncloud.org, user: test, password: test)
No. 

### Environment data
Android version: Oxygen OS 9.0.9

Device model: Oneplus 6  (A 6003 )

Stock or customized system: Stock

ownCloud app version: 2.15 release 4984047c

ownCloud server version: 1.4.1.3

#### Web server error log
There's nothing added to the server log upon upload failure. 
```
Insert your webserver log here
```

#### ownCloud log (data/owncloud.log)
Attached is the log from the app detailing the error. The bulk of it seems to be around line 2290
[ownCloud.log](https://github.com/owncloud/android/files/4902399/ownCloud.log)

**Comment by jesmrec** [comment]
I can't reproduce this, and you couldn't reproduce with the demo server, as you reported. Did you check server logs? 

> ownCloud server version: 1.4.1.3

is that correct? do you mean 10.4.1.3?

**Comment by Kraksna83** [comment]
Apologies, the server version is 10.4.1 ( stable ) 
nothing appears in the server log.  When i upload something, the only error is in the android app. 
In the meantime, the apache web server log looks like this : 
```
86.49.246.36 - - [14/Jul/2020:10:54:53 +0000] "PROPFIND /remote.php/dav/files/tomas/ HTTP/1.1" 301 619 "-" "Mozilla/5.0 (Android) ownCloud-android/2.15"
86.49.246.36 - tomas [14/Jul/2020:10:54:53 +0000] "PROPFIND /remote.php/dav/files/tomas/ HTTP/1.1" 207 1657 "-" "Mozilla/5.0 (Android) ownCloud-android/2.15"
86.49.246.36 - - [14/Jul/2020:10:54:53 +0000] "PROPFIND /remote.php/dav/files/tomas/IMG_20200708_213044.jpg HTTP/1.1" 301 664 "-" "Mozilla/5.0 (Android) ownCloud-android/2.15"
86.49.246.36 - tomas [14/Jul/2020:10:54:53 +0000] "PROPFIND /remote.php/dav/files/tomas/IMG_20200708_213044.jpg HTTP/1.1" 404 879 "-" "Mozilla/5.0 (Android) ownCloud-android/2.15"
86.49.246.36 - - [14/Jul/2020:10:54:53 +0000] "MKCOL /remote.php/dav/uploads/tomas/cc49bedab8c0d9d1e4f0bf6772103f431594724091004 HTTP/1.1" 301 712 "-" "Mozilla/5.0 (Android) ownCloud-android/2.15"
86.49.246.36 - tomas [14/Jul/2020:10:54:54 +0000] "MKCOL /remote.php/dav/uploads/tomas/cc49bedab8c0d9d1e4f0bf6772103f431594724091004 HTTP/1.1" 201 568 "-" "Mozilla/5.0 (Android) ownCloud-android/2.15"
86.49.246.36 - - [14/Jul/2020:10:54:54 +0000] "PUT /remote.php/dav/uploads/tomas/cc49bedab8c0d9d1e4f0bf6772103f431594724091004/0 HTTP/1.1" 301 716 "-" "Mozilla/5.0 (Android) ownCloud-android/2.15"
86.49.246.36 - tomas [14/Jul/2020:10:54:54 +0000] "PUT /remote.php/dav/uploads/tomas/cc49bedab8c0d9d1e4f0bf6772103f431594724091004/0 HTTP/1.1" 201 601 "-" "Mozilla/5.0 (Android) ownCloud-android/2.15"
86.49.246.36 - - [14/Jul/2020:10:54:55 +0000] "PUT /remote.php/dav/uploads/tomas/cc49bedab8c0d9d1e4f0bf6772103f431594724091004/1 HTTP/1.1" 301 716 "-" "Mozilla/5.0 (Android) ownCloud-android/2.15"
86.49.246.36 - tomas [14/Jul/2020:10:54:56 +0000] "PUT /remote.php/dav/uploads/tomas/cc49bedab8c0d9d1e4f0bf6772103f431594724091004/1 HTTP/1.1" 201 601 "-" "Mozilla/5.0 (Android) ownCloud-android/2.15"
86.49.246.36 - - [14/Jul/2020:10:54:56 +0000] "PUT /remote.php/dav/uploads/tomas/cc49bedab8c0d9d1e4f0bf6772103f431594724091004/2 HTTP/1.1" 301 716 "-" "Mozilla/5.0 (Android) ownCloud-android/2.15"
86.49.246.36 - tomas [14/Jul/2020:10:54:57 +0000] "PUT /remote.php/dav/uploads/tomas/cc49bedab8c0d9d1e4f0bf6772103f431594724091004/2 HTTP/1.1" 201 601 "-" "Mozilla/5.0 (Android) ownCloud-android/2.15"
86.49.246.36 - - [14/Jul/2020:10:54:57 +0000] "PUT /remote.php/dav/uploads/tomas/cc49bedab8c0d9d1e4f0bf6772103f431594724091004/3 HTTP/1.1" 301 716 "-" "Mozilla/5.0 (Android) ownCloud-android/2.15"
86.49.246.36 - tomas [14/Jul/2020:10:54:58 +0000] "PUT /remote.php/dav/uploads/tomas/cc49bedab8c0d9d1e4f0bf6772103f431594724091004/3 HTTP/1.1" 201 601 "-" "Mozilla/5.0 (Android) ownCloud-android/2.15"
```
I have enabled the debug of the webserver
```<?php
$CONFIG = array (
  'debug' => true,
etc..
```

but still,  there's nothing new in the owncloud.log on the server itself.

**Comment by Kraksna83** [comment]
[oc.log](https://github.com/owncloud/android/files/5026700/oc.log)
I managed to enable debugging on the server and it actually logs something now. Here's the log from upload attempt.

**Comment by jesmrec** [comment]
Will take a look to the logs. thanks!

**Comment by jesmrec** [comment]
@Kraksna83 do you have logs from the Android side? to enable logs, you have to go to Settings, click 5 times in the latest cell (ownCloud for Android). Then `Logs` section will be enabled. I'd be useful if you could clean the logs before reproducing the problem, so the report will be scoped.

**Comment by Kraksna83** [comment]
Does it need to be from the same event  ? The log I uploaded in the very first post is from the app, but obviously, it's a different upload event.

If not, i'll just make a new log.

**Comment by jesmrec** [comment]
@Kraksna83 we have just released 2.15.1. Could you take a look if the problem keeps being reproducible?

**Comment by Kraksna83** [comment]
@jesmrec 
Unfortunately, It's still the same.  Here's the app log from the new version, there should be 2 attempts at an upload. 
[ownCloud-2-15.1.log](https://github.com/owncloud/android/files/5064810/ownCloud-2-15.1.log)

**Comment by Kraksna83** [comment]
update to latest stable of the server didn't have any effect..

**Comment by jesmrec** [comment]
This is pretty strange. We can take a deeper look with a test account on your server. Would it be posible

**Comment by jesmrec** [comment]
Thanks for sending the test account. 

I have uploaded several files by using the camera, the document provider etc... and i did not get any error. 

Most posible cause is `Android version: Oxygen OS 9.0.9` : The OnePlus operating system could be plenty related with the problem. Unfortunately, i do not own this kind of device to test.

We will have to check the logs deeply.

**Comment by Kraksna83** [comment]
I have updated the android to the latest stable oxygenos 10.3.5 and re-tested the upload.  Still fails, error message in the logs doesn't seem to change : 

`

V: 2020-09-22 17:54:58:764(ChunkFromFileRequestBody.java:87) .writeTo()Sink buffer size: 0
V: 2020-09-22 17:54:58:765(ChunkFromFileRequestBody.java:91) .writeTo()Read 4096 bytes from file channel to java.nio.HeapByteBuffer[pos=4096 lim=4096 cap=4096]
V: 2020-09-22 17:54:58:765(ChunkFromFileRequestBody.java:97) .writeTo()Write 4096 bytes to sink buffer with size 0
V: 2020-09-22 17:54:58:765(ChunkFromFileRequestBody.java:87) .writeTo()Sink buffer size: 0
V: 2020-09-22 17:54:58:765(ChunkFromFileRequestBody.java:91) .writeTo()Read 4096 bytes from file channel to java.nio.HeapByteBuffer[pos=4096 lim=4096 cap=4096]
E: 2020-09-22 17:54:58:766(ChunkFromFileRequestBody.java:114) .writeTo()java.net.ProtocolException: expected 678832 bytes but received 679936
	at okhttp3.internal.connection.Exchange$RequestBodySink.write(Exchange.kt:214)
	at okio.RealBufferedSink.flush(RealBufferedSink.kt:247)
	at com.owncloud.android.lib.common.network.ChunkFromFileRequestBody.writeTo(ChunkFromFileRequestBody.java:95)
	at okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.kt:59)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:100)
`

**Comment by Kraksna83** [comment]
TLDR :  It seems the issue is resolved and can be closed.  

I know nothing about Android app development, but I wanted to fiddle with the part of the code which throws the exception. Maybe add a print statement here and there, wait statement or something which could provide some insight on what's wrong. I downloaded Android studio, git cloned the repo, installed android 29 SDK and some other stuff. Removed the app from the phone and ran it via the android studio built straight from the source code. Well.... to my surprise, it worked. Then, I built an APK from the source and installed it on a phone via ADB. It appears to be working now, so while i still don't know what the issue was, I have no chance of finding out. 

attached is the build which works for me, but please note I have changed NOTHING in the source and just built it. 

[owncloud_2.15.2-debug.zip](https://github.com/owncloud/android/files/5266431/owncloud_2.15.2-debug.zip)

**Comment by jesmrec** [comment]
> edit: @jsmrec the testing user is now removed, because it appears meaningless now.

👍 

Then, we can close this. Thanks for your engagement!. Feel free to reopen if problem persists or new problems raise up
